### ISSUE TITLE
App crash on large number of errors

### ISSUE THREAD

**Issue by DaniilSokolyuk** [JeringTech/Javascript.NodeJS#54]
When a heavy load and a large number of errors from node.js, the application crashes with an error
```
Index was outside the bounds of the array.
   at System.String.get_Chars(Int32 index)
   at Jering.Javascript.NodeJS.OutOfProcessNodeJSService.TryCreateMessage(StringBuilder stringBuilder, String data, String& message)
   at Jering.Javascript.NodeJS.OutOfProcessNodeJSService.ErrorDataReceivedHandler(Object sender, DataReceivedEventArgs evt)
   at System.Diagnostics.Process.ErrorReadNotifyUser(String data)
   at System.Diagnostics.AsyncStreamReader.FlushMessageQueue(Boolean rethrowInNewThread)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Diagnostics.AsyncStreamReader.<>c.<FlushMessageQueue>b__18_0(Object edi)
   at System.Threading.QueueUserWorkItemCallback.<>c.<.cctor>b__6_0(QueueUserWorkItemCallback quwi)
   at System.Threading.ExecutionContext.RunForThreadPoolUnsafe[TState](ExecutionContext executionContext, Action`1 callback, TState& state)
   at System.Threading.QueueUserWorkItemCallback.Execute()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
```

**Comment by JeremyTCD** [comment]
Do you know what your JS is writing to console?  

`TryCreateMessage` processes stdout, accumulating characters until '\0':

https://github.com/JeringTech/Javascript.NodeJS/blob/fbd57f7633ab0aef831eb69bf433c4610ad1fa50/src/NodeJS/NodeJSServiceImplementations/OutOfProcess/OutOfProcessNodeJSService.cs#L320-L339

https://github.com/JeringTech/Javascript.NodeJS/blob/fbd57f7633ab0aef831eb69bf433c4610ad1fa50/src/NodeJS/NodeJSServiceImplementations/OutOfProcess/OutOfProcessNodeJSService.cs#L281-L305

Will try to reproduce in unit tests tomorrow, if you know what your applications is writing stdout or you can debug and check what the data argument is on crashes it'll help.

**Comment by JeremyTCD** [comment]
I've updated my previous comment, I think calling `console.log(undefined)` or `console.log(null)` may cause the error. Need to test and add guards.

**Comment by DaniilSokolyuk** [comment]
Some error from workerpool library when terminate pool, data is empty
![image](https://user-images.githubusercontent.com/2946286/69497558-656f7f80-0eef-11ea-9513-c636160629c4.png)

**Comment by JeremyTCD** [comment]
Thanks for the extra info. #55 handles the case where data is an empty string. 

I'd still like to figure out why empty strings are being written to stdout/stderr by Node.js. Turns out  `console.log(null)`, `console.log(undefined)` and `console.log(obj)` work fine.

Could you share how I can reproduce your situation?

**Comment by JeremyTCD** [comment]
Turns out data is empty when log messages contain an empty line, e.g `console.log('a\n\nb')`. `TryCreateMessage` wasn't handling empty lines properly.  

In your case, most probably some library called `console.error` with a detailed message containing empty lines for formatting.  

Thanks for reporting this bug. I've released v4.4.1 with the fix. Feel free to close this issue if it resolves the problem for you.
